dvc/_debug.py

```diff
@@ -196,35 +196,83 @@ def debugtools(args: Optional["Namespace"] = None, **kwargs):
 
 
 def add_debugging_flags(parser):
+    import os
     from argparse import SUPPRESS
 
-    parser.add_argument("--cprofile", action="store_true", default=False, help=SUPPRESS)
-    parser.add_argument("--yappi", action="store_true", default=False, help=SUPPRESS)
+    # For detailed info see:
+    # https://github.com/iterative/dvc/wiki/Debugging,-Profiling-and-Benchmarking-DVC
+    dvc_show_debug_options = os.environ.get("DVC_SHOW_DEBUG_OPTIONS", "")
```

Comment:
Seems a bit too verbose. Maybe better to use `--debug` flag? Do you know maybe some examples of how other projects handle this?

Reply:
In the long term, I'd really like to introduce the concept of "extensions", even if it's for our own use. We could enable extra functionalities if the `devel` extension is enabled, for example, in the config.

Reply:
I tend to opt for verbosity when I'm unsure. I think `DVC_DEBUG=1 <cmd>` would likely work as well.

I don't see an easy way to add a `--debug` option because the argparse object hasn't finished being constructed at the time where we need to determine if we use SUPRESS or add a help message. I think the environment variable approach is the cleanest way to allow these options to be conditionally exposed in `--help`.

The other options I can think of are:

* Add a `--debug` flag which prints a special help section, but that would need to be maintained separately, so that's undesirable. 

* Check if `"--debug" in sys.argv`, but that's undesirable because 1. the parser might not be applied to sys.argv, and 2. there is a chance `--debug` might appear on the command line, but not be intended as an argument flag (e.g. `dvc add -- --debug`). Checking `sys.argv` won't respect the `--` like argparse will. These issues probably won't arise in practice, and even if they did it would likely not matter because it just changes the content of the help message. Still, I think it is unclean design and the environment variable approach seems like the "idomatic" thing to do here. That being said,  `"--debug" in sys.argv` is simple, and I'm open to doing it this way. The only thing I think is important is that there is a way for a developer to unhide these options.

Reply:
@skshetry Seems like a total overkill for this purpose. But I'm not even sure what extensions you mean, I suppose like git extensions? Don't quite see the need.

Reply:
@Erotemic DVC_DEBUG sounds fine to me, let's not overthink it.

Reply:
@Erotemic Also note dvc/env.py where we try to keep all env vars in.

Reply:
@Erotemic regarding the cli option, there is a way to partially parse known options in argparse IIRC. Just checking argv (not sys.argv) is also fine.

Reply:
Switched to DVC_DEBUG and added it to env.

I do like the idea of showing the extra help with `vebose` is set. At first I wasn't sure how to do it, but I think its possible as long as the parser is initialized with `add_help=False` (which it is) and if we can ensure the verbose argument is added to the parser before we call `add_debugging_flags`, which should just require reordering the code in `get_parent_parser`.

Now that I see how to do this, I think it would be a good idea to remove the DVC_DEBUG environment variable entirely, and just add the extra help options is `-v` is already given (which will make it much easier for a developer to stumble on this too, which is good).

Reply:
I made the above change, I can revert if you want to go with the environ, but I think just respecting `-v` is much cleaner now that I know about the `add_help` argment to `ArgumentParser`.

Reply:
Thank you @Erotemic and @skshetry 🙏